src/client/QXmppPubSubManager.h

```diff
@@ -52,14 +60,18 @@ class QXMPP_EXPORT QXmppPubSubManager : public QXmppClientExtension
     ~QXmppPubSubManager();
 
     // Generic PubSub (the PubSub service is the given entity)
+    QFuture<FeaturesResult> requestFeatures(const QString &serviceJid, bool onlyForPepService = false);
     QFuture<NodesResult> fetchNodes(const QString &jid);
     QFuture<Result> createNode(const QString &jid, const QString &nodeName);
     QFuture<Result> createNode(const QString &jid, const QString &nodeName, const QXmppPubSubNodeConfig &config);
     QFuture<InstantNodeResult> createInstantNode(const QString &jid);
     QFuture<InstantNodeResult> createInstantNode(const QString &jid, const QXmppPubSubNodeConfig &config);
     QFuture<Result> deleteNode(const QString &jid, const QString &nodeName);
+    QFuture<ItemIdsResult> requestItemIds(const QString &serviceJid, const QString &nodeName);
+    template<typename T = QXmppPubSubItem>
+    QFuture<ItemResult<T>> requestItem(const QString &jid, const QString &nodeName, const QString &itemId);
     template<typename T = QXmppPubSubItem>
-    QFuture<ItemResult<T>> requestItem(const QString &jid, const QString &nodeName, const QString &itemId = PUBSUB_CURRENT_ITEM);
+    QFuture<ItemResult<T>> requestItem(const QString &jid, const QString &nodeName, ItemId itemId = Current);
```

Comment:
i'm not 100%-sure about the default parameter here, it's probably very convenient, but I could also imagine cases where it's not so clear what a `requestItem(jid, node)` call does.

Reply:
I think that it is a good default because there are several use cases where one node does only contain one item.

Reply:
OK

Reply:
I'm sorry if I change my mind on this, but I don't think that `requestItem(jid, node)` is clear if you're reading the code without additional knowledge about `requestItem`. (`requestCurrentItem(jid, node)` would make clear which item (but the name and the concept have other issues)).

If you think `QXmppPubSubManager::Current` is too long (which I can understand), you can of course use something like `using PubSub = QXmppPubSubManager;` and then `requestItem(jid, node, PubSub::Current)`.

I don't want to exclude that we add the default parameter later on. That's an option and absolutely no problem (but removing it later would be problematic).

Reply:
I do not think that `QXmppPubSubManager::Current` is too long. But I think that methods with `StandardItemId` as a parameter could have had reasonable defaults. There are several locations where the default value could have been used by the upcoming OMEMO manager. If you make such changes in the future, please make sure to refactor it correctly and consistently.